src/customAuthentication/CustomAuthentication.ts

```diff
@@ -0,0 +1,179 @@
+import { Signature } from '../index';
+import jwt from 'jsonwebtoken';
+import { P12ReaderData } from '../P12Reader';
+interface DecodedToken {
+  valor: string;
+  timestamp: string;
+  iat: number;
+  exp: number;
+}
+/**
+ * Custom Authentication class for generating and verifying tokens
+ */
+class CustomAuthentication {
+  private cert: P12ReaderData;
+
+  constructor(cert: P12ReaderData) {
+    if (!cert.key || !cert.publicKey) {
+      throw new Error('Certificate key and Pem are not defined');
+    }
+    this.cert = cert;
+  }
+
+  generateSeed() {
+    // Generate base64 encoded random value
+    const randomBytes = crypto.randomBytes(128);
+    const randomValue = randomBytes.toString('base64');
+
+    // Generate formatted date with timezone offset
+    const date = new Date();
+    const offset = -4; // UTC-4 for Dominican Republic
+    const localDate = new Date(date.getTime() + offset * 3600 * 1000);
+    const formattedDate = localDate.toISOString().replace('Z', '-04:00');
+
+    const sampleXml = `<?xml version="1.0" encoding="utf-8"?>
+    <SemillaModel xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+        <valor>${randomValue}</valor>
+        <fecha>${formattedDate}</fecha>
+    </SemillaModel>`;
+
+    return sampleXml;
+  }
+
+  async verifySignedSeed(signedXml: string): Promise<string> {
+    try {
+      const doc = new DOMParser().parseFromString(signedXml, 'application/xml');
+
+      // Extract elements
+      const valorNode = xpath.select1('string(//SemillaModel/valor)', doc);
+      const digestValue = xpath.select1(
+        "string(//*[local-name(.)='DigestValue'])",
+        doc
+      );
+      const signatureValue = xpath.select1(
+        "string(//*[local-name(.)='SignatureValue'])",
+        doc
+      );
+
+      if (!valorNode || !digestValue || !signatureValue) {
+        throw new Error('Required XML elements not found');
+      }
+
+      // Calculate digest with updated canonicalization
+      const calculatedDigest = await this.canonicalizeXmlFn(signedXml);
+
+      const expectedDigest = digestValue.toString().trim();
+
+      if (calculatedDigest !== expectedDigest) {
+        throw new Error(
+          `Digest mismatch.\nCalculated: ${calculatedDigest}\nExpected: ${expectedDigest}`
+        );
+      }
+
+      if (!this.cert.key) {
+        throw new Error('Certificate key is not defined');
+      }
+      // Generate bearer token if validation passes
+      const token = jwt.sign(
+        {
+          valor: valorNode.toString(),
+          timestamp: new Date().toISOString(),
+        },
+        this.cert.key,
+        {
+          algorithm: 'RS256',
+          expiresIn: '1h',
+        }
+      );
+
+      return token;
+    } catch (error) {
+      if (error instanceof Error) {
+        throw new Error(`Verification failed: ${error?.message}`);
+      } else {
+        throw new Error(`Verification failed: ${error}`);
+      }
```

Comment:
Perhaps this would work too

```suggestion
      throw new Error(`Verification failed: ${error?.message ?? error}`);
```

Reply:
it should but TS error type is not define and I'm getting `Property 'message' does not exist on type '{}'.ts(2339)` and I'm trying to avoid to use `any` `if(error: any)`